### DISCUSSION TITLE
Validation - Customization of client side validation

### DISCUSSION THREAD

**Discussion by rajenindianic** [filamentphp/filament#15216] [Help] [answered]
### Package

Form builder

### Package Version

v3.2

### How can we help you?

Hi,

I need to customize the client side validation, which is showing in brand name field in below image, I want to show this client side validation below input field.

![Screenshot from 2024-12-31 21-08-28](https://github.com/user-attachments/assets/da2d805b-746e-4043-a121-b27d7fa840de)

Below is my current code.

```
TextInput::make('name')
 ->label('Brand Name')
 ->maxLength(20)
 ->required()
 ->rules(['required'])
 ->unique(ignoreRecord: true),
```
Thank You.

**Comment by leandrocfe** [comment, marked as answer]
->required() shows the browser validation first, as you mentioned.
->rule() shows the server validation. Maybe you can achieve what you want, but I think you need to create a custom field and develop it

**Comment by rajenindianic** [comment]
@leandrocfe Thank You.
